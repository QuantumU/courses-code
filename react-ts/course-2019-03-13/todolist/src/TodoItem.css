.todo {
  margin: 5px;
  padding: 5px;
  background-color: lightblue;
}

.todo.completed {
  background-color: lightgrey;
  text-decoration: line-through;
}