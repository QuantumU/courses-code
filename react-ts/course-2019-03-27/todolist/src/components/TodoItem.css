.todo-item {
  background-color: lightgreen;
  padding: 8px;
  margin: 8px;
}

.todo-item.completed {
  background-color: lightgrey;
  text-decoration: line-through;
}
