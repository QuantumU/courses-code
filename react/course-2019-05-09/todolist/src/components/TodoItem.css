.todo {
  background-color: lightblue;
  margin: 5px;
  padding: 5px;
  list-style-type: none;
}

.todo.completed {
  background-color: grey;
  text-decoration: line-through;
}
