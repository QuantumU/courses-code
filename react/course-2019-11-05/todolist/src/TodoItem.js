import React from "react";

import styles from "./TodoItem.module.css";

function TodoItem({ todo, toggleTodo, deleteTodo }) {
  if (!todo || !todo.title) {
    throw new Error("invalid / missing prop 'todo'");
  }
  const classNames = [styles.todoItem];
  if (todo.completed) {
    classNames.push(styles.completed);
  }
  const className = classNames.join(" ");
  return (
    <div className={className} onClick={() => toggleTodo(todo.id)}>
      {todo.completed ? "DONE: " : "TODO: "}{todo.title}
      <button onClick={() => deleteTodo(todo.id)}>X</button>
    </div>
  );
}

export default TodoItem;
