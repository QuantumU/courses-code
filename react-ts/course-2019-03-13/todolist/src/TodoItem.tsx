import React, { memo, Component } from 'react';
import TodoType from './TodoType';

import './TodoItem.css';

type TodoItemProps = {
  todo: TodoType;
  onCompleted: () => void;
};

class TodoItem extends Component<TodoItemProps, {}> {

  shouldComponentUpdate(newProps: TodoItemProps) {
    return this.props.todo !== newProps.todo;
  }

  render = () => {
    return (
      <div
        className={`todo${this.props.todo.completed ? ' completed' : ''}`}
        onClick={() => {
          this.props.onCompleted();
        }}
      >
        {this.props.todo.completed ? 'DONE: ' : 'TODO: '}{this.props.todo.title}
      </div>
    );
  };
}

export default TodoItem;
