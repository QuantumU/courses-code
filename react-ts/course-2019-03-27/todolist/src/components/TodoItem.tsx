import React, { PureComponent } from 'react';
import TodoType from '../TodoType';
import './TodoItem.css';

type TodoItemProps = {
  todo: TodoType;
  onToggle: (id: number) => void;
};

type TodoItemState = {};

class TodoItem extends PureComponent<TodoItemProps, TodoItemState> {
  render() {
    // "todo-item" or "todo-item completed"
    return (
      <div
        className={
          this.props.todo.completed ? 'todo-item completed' : 'todo-item'
        }
        id="abc"
        onClick={() => {
          this.props.onToggle(this.props.todo.id);
        }}
      >
        {this.props.todo.title}
      </div>
    );
  }
}

export default TodoItem;
