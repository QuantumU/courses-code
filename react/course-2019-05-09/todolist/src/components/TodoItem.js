import React, { Component } from "react";
import PropTypes from "prop-types";

import "./TodoItem.css";

class TodoItem extends Component {
  render() {
    // this.props.todo.id
    // this.props.todo.completed
    return (
      <li
        className={this.props.todo.completed ? "todo completed" : "todo"}
        onClick={() => {
          this.props.onToggle(this.props.todo.id);
        }}
      >
        {this.props.todo.completed ? "DONE:" : "TODO:"} {this.props.todo.title}
      </li>
    );
  }
}

TodoItem.propTypes = {
  todo: PropTypes.shape({
    id: PropTypes.number.isRequired,
    title: PropTypes.string.isRequired,
    completed: PropTypes.bool.isRequired
  }),
  onToggle: PropTypes.func
};

export default TodoItem;
