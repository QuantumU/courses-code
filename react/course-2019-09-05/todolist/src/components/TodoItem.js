import React from "react";
import "./TodoItem.css";

function TodoItem({ todo, onToggle }) {
  const prefix = todo.completed ? "DONE: " : "TODO: ";
  const textContent = prefix + todo.title;

  const handleClick = () => {
    onToggle(todo.id);
  };

  if (todo.completed) {
    return (
      <div className="todo" onClick={handleClick}>
        <del>{textContent}</del>
      </div>
    );
  } else {
    return (
      <div className="todo" onClick={handleClick}>
        {textContent}
      </div>
    );
  }
}

export default TodoItem;
