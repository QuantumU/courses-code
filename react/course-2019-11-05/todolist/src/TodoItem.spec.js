import React from "react";
import TestRenderer from "react-test-renderer";

import TodoItem from "./TodoItem";

it("renders a TodoItem without crashing", () => {
  const instance = TestRenderer.create(
    <TodoItem todo={{ title: "test", completed: true }} />
  ).root;
});

describe("rendering", () => {
  it("renders a div starting with 'TODO:'", () => {
    const instance = TestRenderer.create(
      <TodoItem todo={{ title: "a", completed: false }} />
    ).root;
    const div = instance.findByType("div");
    const firstTextNode = div.children[0];
    expect(firstTextNode).toEqual("TODO: ");
  });

  it("renders a div containing the todo title", () => {
    const renderedTitle = "abc";
    const instance = TestRenderer.create(
      <TodoItem todo={{ title: renderedTitle, completed: false }} />
    ).root;
    const secondNode = instance.findByType("div").children[1];
    expect(secondNode).toEqual(renderedTitle);
  });
});

describe("events", () => {
  it("fires a toggleTodo event if the todo is clicked", () => {
    const mockFn = jest.fn();
    const instance = TestRenderer.create(
      <TodoItem
        todo={{ id: 1, title: "a", completed: false }}
        toggleTodo={mockFn}
      />
    ).root;
    // trigger the click event on the div in the instance
    const divNode = instance.findByType("div");
    divNode.props.onClick();
    // check if mockFn was called with id 1
    expect(mockFn).toHaveBeenCalledWith(1);
  });
});

describe("errors", () => {
  it("throws if title was not provided", () => {
    const functionThatShouldThrow = () => {
      TestRenderer.create(<TodoItem todo={{ id: 1, completed: false }} />);
    };
    expect(functionThatShouldThrow).toThrow("invalid / missing prop 'todo'");
  });
});

it("matches snapshot", () => {
  const tree = TestRenderer.create(
    <TodoItem todo={{ id: 1, completed: true, title: "a" }} />
  ).toJSON();
  expect(tree).toMatchSnapshot();
});
