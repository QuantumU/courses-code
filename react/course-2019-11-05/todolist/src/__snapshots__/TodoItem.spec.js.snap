// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div
  className="todoItem completed"
  onClick={[Function]}
>
  DONE: 
  a
  <button
    onClick={[Function]}
  >
    X
  </button>
</div>
`;
